src/util/detect-bugs.js

```diff
@@ -1,23 +1,65 @@
-import TRANSITIONRUN_EVENT_LOOP_BUG from "./detect-bugs/transitionrun-loop.js";
-import UNREGISTERED_TRANSITION_BUG from "./detect-bugs/unregistered-transition.js";
+import detectTransitionRunLoopBug from "./detect-bugs/transitionrun-loop.js";
+import detectUnregisteredTransitionBug from "./detect-bugs/unregistered-transition.js";
 
-/**
- * Data structure for all detected bugs.
- * All bugs start off as true, and once their promises resolve, that is replaced with the actual value
- */
-export const bugs = {
-	TRANSITIONRUN_EVENT_LOOP: true,
-	UNREGISTERED_TRANSITION: true,
-};
+export default class Bugs {
+	/**
+	 * Whether the browser has the transition run loop bug
+	 * @type {boolean | null}
+	 * @private
+	 */
+	static #transitionRunLoop = null;
 
-TRANSITIONRUN_EVENT_LOOP_BUG.then(value => {
-	bugs.TRANSITIONRUN_EVENT_LOOP = value;
-});
+	/**
+	 * Whether the browser has the unregistered transition bug
+	 * @type {boolean | null}
+	 * @private
+	 */
+	static #unregisteredTransition = null;
 
-UNREGISTERED_TRANSITION_BUG.then(value => {
-	bugs.UNREGISTERED_TRANSITION = value;
-});
+	/**
+	 * Detect all bugs in parallel
+	 * @returns {Promise<{ transitionRunLoop: boolean, unregisteredTransition: boolean }>}
+	 */
+	static async detect () {
+		const [transitionRunLoop, unregisteredTransition] = await Promise.all([
+			detectTransitionRunLoopBug(),
+			detectUnregisteredTransitionBug(),
+		]);
 
-export { TRANSITIONRUN_EVENT_LOOP_BUG, UNREGISTERED_TRANSITION_BUG };
-export const detected = Promise.all([TRANSITIONRUN_EVENT_LOOP_BUG, UNREGISTERED_TRANSITION_BUG]);
-export default bugs;
+		this.#transitionRunLoop = transitionRunLoop;
+		this.#unregisteredTransition = unregisteredTransition;
+
+		return {
+			transitionRunLoop,
+			unregisteredTransition,
+		};
+	}
+
+	/**
+	 * Whether the browser has the transition run loop bug
+	 * @type {boolean}
+	 */
+	static get transitionRunLoop () {
+		if (this.#transitionRunLoop === null) {
+			this.#transitionRunLoop = false; // Default to false while detecting
+			detectTransitionRunLoopBug().then(result => {
+				this.#transitionRunLoop = result;
+			});
+		}
+		return this.#transitionRunLoop;
+	}
```

Comment:
Is there a chance we might face race conditions here? 🤔

Reply:
Well, this certainly doesn't always return the correct result the first time it's called (assuming `detect()` wasn't called or didn't complete yet), but [that was _also_ the case before this PR](https://github.com/LeaVerou/style-observer/blob/main/src/util/detect-bugs.js#L6). I just noticed that the bugs defaulted to `true` instead of `false` though; I will update this PR to keep that behavior.

Reply:
Done in https://github.com/LeaVerou/style-observer/pull/111/commits/36096aa70ea21f1f970f32cbbcfe271d9bbe6813. Let me know if you think we should use a different approach!

Reply:
> I just noticed that the bugs defaulted to `true` instead of `false` though

IIRC, we did it because there is no harm in doing some extra work, even if a bug is not present, than not doing anything if it is.

Reply:
If that is the case, then there shouldn't be any possible race conditions, unless I'm missing something.